diff --git a/vendor/magento/module-search/Model/ResourceModel/SynonymReader.php b/vendor/magento/module-search/Model/ResourceModel/SynonymReader.php
index ff72c30..4ce110f 100644
--- a/vendor/magento/module-search/Model/ResourceModel/SynonymReader.php
+++ b/vendor/magento/module-search/Model/ResourceModel/SynonymReader.php
@@ -88,7 +88,7 @@ class SynonymReader extends AbstractDb
         $matchQuery = $this->fullTextSelect->getMatchQuery(
             ['synonyms' => 'synonyms'],
             $phrase,
-            Fulltext::FULLTEXT_MODE_BOOLEAN
+            Fulltext::FULLTEXT_MODE_NATURAL
         );
         $query = $this->getConnection()->select()->from(
             $this->getMainTable()
